owncloudApp/src/androidTest/java/com/owncloud/android/settings/camerauploads/OCSettingsCameraUploadsTest.kt

```diff
@@ -0,0 +1,255 @@
+        //Asserts
+        assertTrue(mPrefCameraPictureUploads.isChecked)
+        //Reset
+        removePictureSubOptions()
+    }
+
+    @Test
+    fun disableVideoUploadsAccept() {
+        enableVideoSubOptions()
+        onView(withText(R.string.prefs_camera_video_upload)).perform(click());
+        onView(withText(R.string.common_yes)).perform(click());
+        //Asserts
+        Assert.assertFalse(mPrefCameraVideoUploads.isChecked)
+        onView(withText(R.string.prefs_camera_video_upload_path_title)).check(doesNotExist())
+        onView(withText(R.string.camera_video_upload_on_wifi)).check(doesNotExist())
+    }
+
+    @Test
+    fun disableVideoUploadsRefuse() {
+        enableVideoSubOptions()
+        onView(withText(R.string.prefs_camera_video_upload)).perform(click());
+        onView(withText(R.string.common_no)).perform(click());
+        //Asserts
+        assertTrue(mPrefCameraVideoUploads.isChecked)
+        //Reset
+        removeVideoSubOptions()
+    }
+
+    @Test
+    fun cameraFolderView() {
+        enablePictureSubOptions()
+        //Asserts
+        onView(withText(String.format(
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title),
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title_required))))
+            .check(matches(isDisplayed()))
+        //Reset
+        removePictureSubOptions()
+    }
+
+    @Test
+    fun cameraOpenPicker() {
+        enablePictureSubOptions()
+        val cameraFolder = Environment.getExternalStoragePublicDirectory(
+            Environment.DIRECTORY_DCIM
+        ).absolutePath + "/Camera";
+        Intents.init()
+        //Asserts
+        onView(withText(String.format(
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title),
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title_required))))
+            .perform(click())
+        Intents.intended(IntentMatchers.hasComponent(LocalFolderPickerActivity::class.java.name))
+        IntentMatchers.hasExtra(LocalFolderPickerActivity.EXTRA_PATH, cameraFolder)
+        Intents.release()
+        onView(withText(android.R.string.cancel)).perform(click())
+        //Reset
+        removePictureSubOptions()
+    }
+
+    @Test
+    fun switchOriginalFileWillBe() {
+        enablePictureSubOptions()
+        onView(withText(R.string.prefs_camera_upload_behaviour_title)).perform(click())
+        onView(withText(R.string.pref_behaviour_entries_move)).perform(click())
+        //Asserts
+        onView(withText(R.string.pref_behaviour_entries_move)).check(matches(isDisplayed()))
+        //Reset
+        onView(withText(R.string.prefs_camera_upload_behaviour_title)).perform(click())
+        onView(withText(R.string.pref_behaviour_entries_keep_file)).perform(click())
+        removePictureSubOptions()
+
+    }
+
+    fun removePictureSubOptions(){
```

Comment:
I would call these methods enableCameraPictureUploads, disableCameraPictureUploads and so on, it would describe better what you try to do.

Reply:
absolutely agreed, done.

Reply:
You renamed them as `enablePictureSubOptions` and `disablePictureSubOptions` but I meant `enableCameraPictureUploads` and `disableCameraPictureUploads`

Reply:
yes, your option is more clear to define the methods. Done.